daft-derive/src/internals/imp.rs

```diff
@@ -694,6 +695,9 @@ impl StructConfig {
         if errors.has_errors() {
             None
         } else {
+            for error in duplicates {
+                errors.push(error);
+            }
```

Comment:
The idea is to bypass `errors.has_errors()` check so we return Some here. It's only nested one level deep so that's enough to get it to produce code:

```rust
fn make_struct_impl(
    input: &DeriveInput,
    s: &DataStruct,
    errors: ErrorSink<'_, syn::Error>,
) -> Option<TokenStream> {
    let Some(struct_config) =
        StructConfig::parse_from(&input.attrs, errors.new_child())
    else {
        // An error occurred parsing the struct configuration -- don't generate
        // anything.
        return None;
    };
```

Though this initial commit fails to add the duplicates back in in the event of a larger error.

Reply:
> Though this initial commit fails to add the duplicates back in in the event of a larger error.

Fixed in the second commit and I added a test exercising this mixed-error case.

The field code is nested much deeper than the struct error code making it non-trivial to apply there. One possible option is introducing some differentiation between the types of errors that are held by `ErrorSink` such as "critical_error" versus "warn_error." So we could have `if errors.has_critical_error() {` instead of `if errors.has_errors() {`

Reply:
Thanks! I see what you mean, and I really like the idea of having two categories of errors the way you've described them. How about updating `ErrorSink` to have "errors" and "warnings", knowing that we have to produce errors for warnings due to proc-macro API limitations.